### PR TITLE
Sql pvup support

### PR DESCRIPTION
**Pull request by usingh83**
If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
sql: added  `password_policy` field to `google_sql_user` resource

```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_sql_user_test.go [hunk 1]
```diff
@@ -311,6 +348,58 @@ resource "google_sql_user" "user2" {
...
+  name     = "admin"
+  instance = google_sql_database_instance.instance.name
+  host     = "google.com"
+  password = "%s"
+  sql_server_user_details {
+    disabled = "%t"
+    server_roles = [ "admin" ]  	
+  }
+	password_policy {
+		allowed_failed_attempts  = 6
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_sql_user_test.go:373]: Spacing here is a little off

File: mmv1/third_party/terraform/tests/resource_sql_user_test.go [hunk 2]
```diff
@@ -279,6 +279,43 @@ func testAccSqlUserDestroyProducer(t *testing.T) func(s *terraform.State) error
 	}
 }
 
+func TestAccSqlUser_mysqlPasswordPolicy(t *testing.T) {
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_sql_user_test.go:282]: Were you able to successfully run this test?

I would be very surprised if this works due to the lack of a flatten method for the new block

Comment (usingh83) [reply mmv1/third_party/terraform/tests/resource_sql_user_test.go:282]: The tests are passing fine, This resource does not have a flatten method for any block. I have followed the same coding style.

Comment (slevenick) [reply mmv1/third_party/terraform/tests/resource_sql_user_test.go:282]: Ah, tests pass fine but this doesn't quite work as we would like it to. Currently the password_policy field is untracked, meaning this resource doesn't read the password_policy field from the API and compare it to state, it just assumes the value in state is present in the API. This makes sense for fields that aren't returned by the API like `password` itself, but doesn't match our normal pattern

We should track it as it does come back in the API response.

Comment (usingh83) [reply mmv1/third_party/terraform/tests/resource_sql_user_test.go:282]: Understood, thanks for explaining, have fixed this and tested the code.

**Comment by slevenick** [comment]
Are you able to run the `TestAccSqlUser_mysqlDisabled` locally and see it pass? I'm seeing a diff after apply during the test run that seems to be caused by the flatten not working correctly

**Comment by slevenick** [comment]
Sorry, I'm not sure how your last change fixes the issue. Have you been able to use this resource locally successfully?

**Review by slevenick — CHANGES_REQUESTED** [review]
See previous comments

**Comment by usingh83** [comment]
> See previous comments

It didn't, i am still working on it.

**Comment by usingh83** [comment]
> Are you able to run the `TestAccSqlUser_mysqlDisabled` locally and see it pass? I'm seeing a diff after apply during the test run that seems to be caused by the flatten not working correctly

Solved it, sorry for the delay, it took me a while to figure this one out.

**Review by slevenick — COMMENTED** [review]
Running this test locally still fails. Have you been able to test this resource locally?

```
          # google_sql_user.user2 will be updated in-place
          ~ resource "google_sql_user" "user2" {
                id       = "admin/gmail.com/i-8532336729755310092"
                name     = "admin"
                # (4 unchanged attributes hidden)

              ~ password_policy {
                  ~ enable_password_verification = false -> true
                    # (3 unchanged attributes hidden)

                  ~ status {
                      ~ locked                   = false -> true
                      ~ password_expiration_time = "2022-10-03T00:14:56Z" -> "2023-07-23T04:06:22.114-07:00"
                    }
                }
            }
```

**Comment by usingh83** [comment]
> Running this test locally still fails. Have you been able to test this resource locally?
> 
> ```
>           # google_sql_user.user2 will be updated in-place
>           ~ resource "google_sql_user" "user2" {
>                 id       = "admin/gmail.com/i-8532336729755310092"
>                 name     = "admin"
>                 # (4 unchanged attributes hidden)
> 
>               ~ password_policy {
>                   ~ enable_password_verification = false -> true
>                     # (3 unchanged attributes hidden)
> 
>                   ~ status {
>                       ~ locked                   = false -> true
>                       ~ password_expiration_time = "2022-10-03T00:14:56Z" -> "2023-07-23T04:06:22.114-07:00"
>                     }
>                 }
>             }
> ```

Hi [slevenick](https://github.com/slevenick), sorry for the late response, I was having a lot of problems with the definitions of some of the fields. I have clarified the fields and found that the `status` block above is going to be a read only block. I have made changes for the same. The tests are passing for me in my local now.

**Comment by usingh83** [comment]
> > Running this test locally still fails. Have you been able to test this resource locally?
> > ```
> >           # google_sql_user.user2 will be updated in-place
> >           ~ resource "google_sql_user" "user2" {
> >                 id       = "admin/gmail.com/i-8532336729755310092"
> >                 name     = "admin"
> >                 # (4 unchanged attributes hidden)
> > 
> >               ~ password_policy {
> >                   ~ enable_password_verification = false -> true
> >                     # (3 unchanged attributes hidden)
> > 
> >                   ~ status {
> >                       ~ locked                   = false -> true
> >                       ~ password_expiration_time = "2022-10-03T00:14:56Z" -> "2023-07-23T04:06:22.114-07:00"
> >                     }
> >                 }
> >             }
> > ```
> 
> Hi [slevenick](https://github.com/slevenick), sorry for the late response, I was having a lot of problems with the definitions of some of the fields. I have clarified the fields and found that the `status` block above is going to be a read only block. I have made changes for the same. The tests are passing for me in my local now.

Please wait with the review, i am trying to see why TestAccSqlUser_mysqlDisabled is failing.

**Comment by usingh83** [comment]
> > > Running this test locally still fails. Have you been able to test this resource locally?
> > > ```
> > >           # google_sql_user.user2 will be updated in-place
> > >           ~ resource "google_sql_user" "user2" {
> > >                 id       = "admin/gmail.com/i-8532336729755310092"
> > >                 name     = "admin"
> > >                 # (4 unchanged attributes hidden)
> > > 
> > >               ~ password_policy {
> > >                   ~ enable_password_verification = false -> true
> > >                     # (3 unchanged attributes hidden)
> > > 
> > >                   ~ status {
> > >                       ~ locked                   = false -> true
> > >                       ~ password_expiration_time = "2022-10-03T00:14:56Z" -> "2023-07-23T04:06:22.114-07:00"
> > >                     }
> > >                 }
> > >             }
> > > ```
> > 
> > 
> > Hi [slevenick](https://github.com/slevenick), sorry for the late response, I was having a lot of problems with the definitions of some of the fields. I have clarified the fields and found that the `status` block above is going to be a read only block. I have made changes for the same. The tests are passing for me in my local now.
> 
> Please wait with the review, i am trying to see why TestAccSqlUser_mysqlDisabled is failing.

Ok now TestAccSqlUser_mysqlDisabled is passing. Don't know why modular-magician did not run.

File: mmv1/third_party/terraform/tests/resource_sql_user_test.go [hunk 3]
```diff
@@ -279,6 +279,43 @@ func testAccSqlUserDestroyProducer(t *testing.T) func(s *terraform.State) error
...
+	// Multiple fine-grained resources
+	skipIfVcr(t)
+	t.Parallel()
+
+	instance := fmt.Sprintf("i-%d", randInt(t))
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_sql_user_test.go:287]: Can you change this to "tf-test-i%d"

We are trying to standardize on the tf-test prefix so we can programmatically sweep resources that get left behind

**Review by slevenick — COMMENTED** [review]
One small note in the test, other than that looks good

File: mmv1/third_party/terraform/tests/resource_sql_user_test.go (hunk 3 above)
Comment (usingh83) [reply mmv1/third_party/terraform/tests/resource_sql_user_test.go]: Done

**Review by slevenick — APPROVED** [review]

**Pull request merged**